src/services/bundle.service.js

```diff
@@ -79,24 +83,37 @@ async function uploadTransactionBundle(req, res) {
       ]
     });
   }
-  checkProvenanceHeader(req.headers);
+  let xprovenanceIncluded;
+  if (Object.keys(req.headers).includes('x-provenance')) {
+    checkProvenanceHeader(req.headers);
+    xprovenanceIncluded = true;
+  }
   const { protocol, baseUrl } = req;
   const scrubbedEntries = replaceReferences(entries);
   const requestsArray = scrubbedEntries.map(async entry => {
     const { url, method } = entry.request;
     const destinationUrl = `${protocol}://${path.join(req.headers.host, baseUrl, baseVersion, url)}`;
-    return axios[method.toLowerCase()](destinationUrl, entry.resource, {
-      headers: { 'Content-Type': 'application/json+fhir', 'X-Provenance': req.headers['x-provenance'] }
-    });
+    if (xprovenanceIncluded) {
+      return axios[method.toLowerCase()](destinationUrl, entry.resource, {
+        headers: { 'Content-Type': 'application/json+fhir', 'X-Provenance': req.headers['x-provenance'] }
+      });
+    } else {
+      return axios[method.toLowerCase()](destinationUrl, entry.resource, {
+        headers: { 'Content-Type': 'application/json+fhir' }
+      });
+    }
```

Comment:
Nitpick: Can we avoid the code duplication here by creating a `headers` variable and then adding the `X-Provenance` key/value only if `xprovenanceIncluded` is true? That way, we can only have one axios call, and just pass in that variable as the headers argument